Tasks/IISWebAppManagement/IISWebAppManagement.ps1

```diff
@@ -0,0 +1,66 @@
+    $hostNameWithOutSNI = Get-VstsInput -Name "HostNameWithOutSNI"
+    $hostNameWithHttp = Get-VstsInput -Name "HostNameWithHttp"
+    $hostNameWithSNI = Get-VstsInput -Name "HostNameWithSNI"
+    $sslCertThumbPrint = Get-VstsInput -Name "SSLCertThumbPrint"
+
+    $createAppPool = Get-VstsInput -Name "CreateAppPool"
+    $appPoolName = Get-VstsInput -Name "AppPoolName"
+    $dotNetVersion = Get-VstsInput -Name "DotNetVersion"
+    $pipeLineMode = Get-VstsInput -Name "PipeLineMode"
+    $appPoolIdentity = Get-VstsInput -Name "AppPoolIdentity"
+    $appPoolUsername = Get-VstsInput -Name "AppPoolUsername"
+    $appPoolPassword = Get-VstsInput -Name "AppPoolPassword"
+    $appCmdCommands = Get-VstsInput -Name "AppCmdCommands"
+    $deployInParallel = Get-VstsInput -Name "DeployInParallel"
+
+    $hostName = Get-HostName -protocol $protocol -hostNameWithHttp $hostNameWithHttp -hostNameWithSNI $hostNameWithSNI -hostNameWithOutSNI $hostNameWithOutSNI -sni $serverNameIndication
+    
+    Trim-Inputs -siteName ([ref]$websiteName) -physicalPath ([ref]$websitePhysicalPath)  -poolName ([ref]$appPoolName) -websitePathAuthuser ([ref]$websiteAuthUserName) -appPoolUser ([ref]$appPoolUsername) -sslCertThumbPrint ([ref]$sslCertThumbPrint)
+
+    Validate-Inputs -createWebsite $createWebsite -websiteName $websiteName -createAppPool $createAppPool -appPoolName $appPoolName -addBinding $addBinding -protocol $protocol -sslCertThumbPrint $sslCertThumbPrint
+    
+    if ($createWebsite -and $websitePhysicalPathAuth -ieq "WebsiteWindowsAuth") {
+        $websiteAuthUserPassword = Escape-SpecialChars -str $WebsiteAuthUserPassword
+        $websitePhysicalPathAuthPassword = "$websiteAuthUserPassword" | ConvertTo-SecureString -AsPlainText -Force
+        $websitePhysicalPathAuthCredentials = New-Object System.Management.Automation.PSCredential ("$websiteAuthUserName", $websitePhysicalPathAuthPassword)
+    }
+
+    if ($createAppPool -and $appPoolIdentity -ieq "SpecificUser") {
+        $appPoolPassword = Escape-SpecialChars -str $appPoolPassword
+        $appPoolPassword = "$appPoolPassword" | ConvertTo-SecureString -AsPlainText -Force
+        $appPoolCredentials = New-Object System.Management.Automation.PSCredential ("$appPoolUsername", $appPoolPassword)
+    }
+
+    $appCmdCommands = Escape-SpecialChars -str $appCmdCommands
+
+    Execute-Main -CreateWebsite $createWebsite -WebsiteName $websiteName -WebsitePhysicalPath $websitePhysicalPath -WebsitePhysicalPathAuth $websitePhysicalPathAuth -websitePhysicalPathAuthCredentials $websitePhysicalPathAuthCredentials -AddBinding $addBinding -Protocol $protocol -IpAddress $ipAddress -Port $port -HostName $hostName -ServerNameIndication $serverNameIndication -SslCertThumbPrint $sslCertThumbPrint -CreateAppPool $createAppPool -AppPoolName $appPoolName -DotNetVersion $dotNetVersion -PipeLineMode $pipeLineMode -AppPoolIdentity $appPoolIdentity -AppPoolCredentials $appPoolCredentials -AppCmdCommands $appCmdCommands
```

Comment:
Why do we need to have physical path credentials ? Doesn't the way in which we configure the machine group agent ensure that we have access to the path?